### PR TITLE
pexports: add pexports/0.47 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **pexports/0.47**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pexports/all/patches/0003-fix-MSVC-x86.patch [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--- pexports.c
++++ pexports.c
+@@ -81,2 +81,2 @@
+-  extern __declspec(dllimport) void __stdcall *GetModuleHandleA(char *name);
+-  extern __declspec(dllimport) void __stdcall *GetProcAddress(void *module, char *name);
++  extern __declspec(dllimport) void * __stdcall GetModuleHandleA(char *name);
++  extern __declspec(dllimport) void * __stdcall GetProcAddress(void *module, char *name);
```
Comment (SSE4) [inline recipes/pexports/all/patches/0003-fix-MSVC-x86.patch:7]: for sure, these functions return pointers (HMODULE/HINSTANCE and a function pointer)
seems like upstream has a bug

File: recipes/pexports/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+if(NOT WIN32)
+    message(FATAL_ERROR "pexports only supports Windows")
+endif()
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
```
Comment (SSE4) [inline recipes/pexports/all/test_package/CMakeLists.txt:8]: why so complex CMakeLists.txt for test_package?
did you borrow it somewhere?

Comment (madebr) [reply recipes/pexports/all/test_package/CMakeLists.txt:8]: No, it's homebrew.
It's just creating a dll -> running pexports on the dll -> running dlltool/lib to create an import library -> create an exe using the import library.

It's do-able outside cmake by just sequentially executing the commands, but I can re-use some cmake variables here.

File: recipes/pexports/all/patches/0003-fix-MSVC-x86.patch (hunk 1 above)
Comment (madebr) [reply recipes/pexports/all/patches/0003-fix-MSVC-x86.patch:7]: I think the C standard says to assume that EAX (=result variable) is clobbered.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/pexports/all/patches/0001-add-help.patch [hunk 3]
```diff
@@ -0,0 +1,32 @@
+--- pexports.c
++++ pexports.c
```
Comment (uilianries) [inline recipes/pexports/all/patches/0001-add-help.patch:2]: it seems be a feature patch, but doesn't break when missing

Comment (madebr) [reply recipes/pexports/all/patches/0001-add-help.patch:2]: Indeed. I added it to be able to test pexports on non-Windows systems.

**Review by uilianries — APPROVED** [review]

**Pull request merged**